File: RPG City Lite Server/src/main/java/com/jbcomputers/rpgcitylite/entities/BuildingAction.java
package com.jbcomputers.rpgcitylite.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@Entity
@Table(name="building_actions")
public class BuildingAction {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name="actions_id")
    Long actionsId;

    @ManyToOne(fetch = FetchType.LAZY, optional=false)
    @JoinColumn(name="buildingid", nullable=false)
    @JsonBackReference
    Building building;

    @Column(name="category")
    String category;

    @Column(name="level")
    int level;

    @Column(name="type")
    String type;

    @Column(name="value")
    int value;
}

File: RPG City Lite Client/RPG City Lite/Assets/Scripts/Entities/BuildingAction.cs

public class BuildingAction 
{
    long actionsId;
    string category;
    int level;
    string type;
    int value;

    public long ActionsId { get => actionsId; set => actionsId = value; }
    public string Category { get => category; set => category = value; }
    public int Level { get => level; set => level = value; }
    public string Type { get => type; set => type = value; }
    public int Value { get => value; set => this.value = value; }
}
